File: GenesisAndroid/src/test/java/com/emerchantpay/gateway/genesisandroid/ErrorCodesUnitTest.java
package com.emerchantpay.gateway.genesisandroid;

import com.emerchantpay.gateway.genesisandroid.api.constants.ErrorCodes;

import org.junit.Before;
import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class ErrorCodesUnitTest {

    private ErrorCodes errorCodes;

    @Before
    public void setup() {
         errorCodes = ErrorCodes.AUTHENTICATION_ERROR;
    }

    @Test
    public void testErrorCodes() {
        assertEquals(errorCodes.getCode(), 110);
        assertEquals(errorCodes.findByCode(110), ErrorCodes.AUTHENTICATION_ERROR);
    }
}

File: GenesisAndroid/src/test/java/com/emerchantpay/gateway/genesisandroid/ErrorCodesUnitTest.kt
package com.emerchantpay.gateway.genesisandroid

import com.emerchantpay.gateway.genesisandroid.api.constants.ErrorCodes

import org.junit.Before
import org.junit.Test

import junit.framework.Assert.assertEquals

class ErrorCodesUnitTest {

    private var errorCodes: ErrorCodes? = null

    @Before
    fun setup() {
        errorCodes = ErrorCodes.AUTHENTICATION_ERROR
    }

    @Test
    fun testErrorCodes() {
        assertEquals(errorCodes!!.getCode(), 110)
        assertEquals(errorCodes!!.findByCode(110), ErrorCodes.AUTHENTICATION_ERROR)
    }
}
